### PR TITLE
Problem: no option to choose the network initialisation (testnet/devnet...) in client-cli / rpc(cro-367)

### PR DESCRIPTION
**Pull request by leejw51crypto**
network initialization option
mainnet
testnet
devnet

### CODE REVIEW

File: chain-core/src/init/network.rs [hunk 1]
```diff
@@ -71,8 +71,8 @@ pub fn get_bech32_human_part() -> &'static str {
 
 mod chosen_network {
     use super::*;
-    pub static mut NETWORK: Network = Network::Devnet;
-    pub static mut NETWORK_ID: u8 = 0 as u8;
+    pub static mut NETWORK: Network = Network::Testnet;
+    pub static mut NETWORK_ID: u8 = 0x42 as u8;
```
Comment (devashishdxt) [inline chain-core/src/init/network.rs]: Is there any specific reason to change these defaults?

File: client-rpc/src/main.rs [hunk 2]
```diff
@@ -37,6 +37,15 @@ pub(crate) struct Options {
     )]
     network_id: String,
 
+    #[structopt(
+        name = "network-type",
+        short = "i",
+        long,
+        default_value = "test",
+        help = "Network Type (main, test, deve)"
```
Comment (devashishdxt) [inline client-rpc/src/main.rs]:
```suggestion
        help = "Network Type (main, test, dev)"
```

File: client-rpc/src/main.rs [hunk 3]
```diff
@@ -37,6 +37,15 @@ pub(crate) struct Options {
     )]
     network_id: String,
 
+    #[structopt(
+        name = "network-type",
+        short = "i",
+        long,
+        default_value = "test",
```
Comment (devashishdxt) [inline client-rpc/src/main.rs]: I think the default value should be `dev`

File: client-rpc/src/server.rs [hunk 4]
```diff
@@ -49,6 +50,13 @@ impl Server {
     pub(crate) fn new(options: Options) -> Result<Server> {
         let network_id =
             hex::decode(&options.network_id).context(ErrorKind::SerializationError)?[0];
+        let network_type = options.network_type;
+        match &network_type[..4] {
+            "main" => init_network(Network::Mainnet),
```
Comment (tomtau) [inline client-rpc/src/server.rs]: `init_network` isn't public anymore.

What about:

1. make `options.network_id` optional argument
2.  use the constants from chain_core and `init_chain_id`: 
- `"main" => init_chain_id(MAINNET_CHAIN_ID)`
- `"test" => init_chain_id(TESTNET_CHAIN_ID)`
- for devnet, require `options.network_id` and do `init_chain_id(format!("devnet-{}", options.network_id))`
3. for the RPC server, use `get_network_id()` from chain-core (after it's initialised)

What do you think?

File: chain-core/src/init/network.rs (hunk 1 above)
Comment (leejw51crypto) [reply chain-core/src/init/network.rs]: no specific reason
devnet is also good.
for devnet
how about 0xab?
@tomtau

File: client-rpc/src/main.rs (hunk 3 above)
Comment (leejw51crypto) [reply client-rpc/src/main.rs]: ok

File: chain-core/src/init/network.rs (hunk 1 above)
Comment (tomtau) [reply chain-core/src/init/network.rs]: just keep it 0, it'll mean that it wasn't initialised

Comment (tomtau) [reply chain-core/src/init/network.rs]: for devnet, anyone can choose any number

File: client-rpc/src/server.rs (hunk 4 above)
Comment (leejw51crypto) [reply client-rpc/src/server.rs]: not bad
so it's like this

--network-type main 
or
--network-type test
or  automaticall dev mode, if network type is not specified
--network-id=ab

File: chain-core/src/init/network.rs (hunk 1 above)
Comment (leejw51crypto) [reply chain-core/src/init/network.rs]: ok, i will change back

**Pull request merged**